MetaMask/metamask-extension
# MV3: Cleanup and update contentscript.js streams

## Explanation

Originally, this PR addressed the "Could not establish connection. Receiving end does not exist." error. It did so with a timeout fix explained in this commit https://github.com/MetaMask/metamask-extension/pull/15939/commits/9ed8f8e5e3be2bcb08a78aa045eb381ef2502410. This commit with the timeout workaround has been removed. We will address this error in this issue: https://github.com/MetaMask/metamask-extension/issues/15970. The error appears to be non-blocking. See related issue for more details.

This PR cleans and updates the `contentscript.js` streams logic for the the dapp provider, legacy provider, and phishing warning page streams: 
- reestablish the page streams on disconnect 
- reorder reset logic by moving `removeAllListeners()` before the `destroy()` calls.
- checks for extension id before attempting to send the keep-alive message 

## More Information

## Screenshots/Screencaps


## Manual Testing Steps

1. Start the extension using MV3 build (`yarn start:mv3`)
2. View dapp page
3. Inspect app-init.js service worker and terminate service worker
4. Observe dapp continues to work

## Pre-Merge Checklist

- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added  - no tests for contentscript.js exist at this time

### + If there are functional changes:
